### ISSUE TITLE
Bug Report: payara-micro-5.2021.5 fails to deploy WAR with "ServletConfig has not been initialized" Exception/FISH-5660

### ISSUE THREAD

**Issue by AndrewG10i** [payara/Payara#5363]
# Description #
----------

The same app can be deployed fine on payara-micro-5.2021.4 but fails to get deployed on payara-micro-5.2021.5 with the following exception:
```
[2021-07-30T10:58:46.945+0800] [] [←[1;91mSEVERE←[0m] [] [←[1;94mfish.payara.micro.PayaraMicro←[0m] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1627613926945] [levelValue: 1000] [[

java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at fish.payara.micro.boot.loader.MainMethodRunner.run(MainMethodRunner.java:50)
        at fish.payara.micro.boot.loader.Launcher.launch(Launcher.java:114)
        at fish.payara.micro.boot.loader.Launcher.launch(Launcher.java:73)
        at fish.payara.micro.boot.PayaraMicroLauncher.create(PayaraMicroLauncher.java:88)
        at fish.payara.micro.boot.PayaraMicroLauncher.main(PayaraMicroLauncher.java:72)
        at fish.payara.micro.PayaraMicro.main(PayaraMicro.java:456)
Caused by: fish.payara.micro.BootstrapException: ServletConfig has not been initialized
        at fish.payara.micro.impl.PayaraMicroImpl.bootStrap(PayaraMicroImpl.java:1070)
        at fish.payara.micro.impl.PayaraMicroImpl.create(PayaraMicroImpl.java:230)
        at fish.payara.micro.impl.PayaraMicroImpl.main(PayaraMicroImpl.java:217)
        ... 10 more
Caused by: java.lang.IllegalStateException: ServletConfig has not been initialized
        at javax.servlet.GenericServlet.getServletContext(GenericServlet.java:159)
        at org.glassfish.jersey.servlet.ServletContainer.getServletContext(ServletContainer.java:457)
        at fish.payara.appserver.rest.endpoints.config.admin.ListRestEndpointsCommand.getEndpointMap(ListRestEndpointsCommand.java:214)
        at fish.payara.micro.impl.PayaraMicroImpl.lambda$dumpFinalStatus$6(PayaraMicroImpl.java:2606)
        at java.util.LinkedHashMap$LinkedValues.forEach(LinkedHashMap.java:608)
        at fish.payara.micro.impl.PayaraMicroImpl.dumpFinalStatus(PayaraMicroImpl.java:2603)
        at fish.payara.micro.impl.PayaraMicroImpl.bootStrap(PayaraMicroImpl.java:1060)
        ... 12 more
]]
```

## Expected Outcome ##

App is to be properly deployed

## Current Outcome ##

Deployment fails with the given above exception, Payara Micro shuts-down.

## Steps to reproduce ## 

1) Download codes from: https://github.com/AndrewG10i/JsfSampleApp
2) Build it: `mvn clean package`
3) This will give the mentioned above exception: `java -jar payara-micro-5.2021.5.jar --deploy jsfDebug-1.0.war`
4) But this works fine: `java -jar payara-micro-5.2021.4.jar --deploy jsfDebug-1.0.war`

As per this sample app seems removing of `src\main\webapp\WEB-INF\glassfish-web.xml` will fix the issue, but in the real production app we have removed it - but exception on deployment still persists. So trying to figure out now how to replicate the same issue in the sample app.

## Environment ##

- **Distribution**: Payara Micro 5.2021.5
- **JDK Version**: 1.8.0_302 Zulu 8.56.0.21-CA-win64
- **Operating System**: Windows 10

**Comment by AndrewG10i** [comment]
So I spent another few hours trying to locate what exactly causes reported exception during deployment in the prod app WAR, besides having `glassfish-web.xml` in the project, but was not able to localize the cause so far.
Any clue would be much appreciated on whether it is some deprecated code in our app or it is an issue with 5.2021.5 Payra Micro.
Thank you!

**Comment by rlfnb** [comment]
running into the same issue with payara-micro-5.2021.5, going back to payara-micro-5.2021.4 runs fine. Might be helpful to find the cause?

**Comment by shub8968** [comment]
Hi @AndrewG10i,

Thanks for bringing it to our notice. I have raised an internal JIRA `FISH-5660` in this regard.

**Comment by phillipross** [comment]
@AndrewG10i I've found that the problem occurs when contextroot is /

I noticed your jsf repo has glassfish web descriptor setting the contextroot to / so it makes sense if you remove the descriptor then the error doesn't occur 😉

**Comment by AndrewG10i** [comment]
> 
> 
> @AndrewG10i I've found that the problem occurs when contextroot is /
> 
> I noticed your jsf repo has glassfish web descriptor setting the contextroot to / so it makes sense if you remove the descriptor then the error doesn't occur 😉

Thanks, I see now: in our app we are building uberjar with the directive like `--deploy $warPath;`, what tells to deploy WAR in context root, thus it is failing...

**Comment by AndrewG10i** [comment]
I do confirm that this issue is resolved in Payara Micro 5.2021.7. Thank you guys!
